delta_go/game_mechanics.py

```diff
@@ -0,0 +1,265 @@
+import pickle
+import random
+import sys
+from pathlib import Path
+from typing import Any, Callable, Dict, Optional, Tuple
+
+import numpy as np
+
+import pygame
+import torch
+from go_base import (
+    BLACK,
+    BOARD_SIZE,
+    PASS_MOVE,
+    WHITE,
+    all_legal_moves,
+    game_over,
+    int_to_coord,
+    is_move_legal,
+    play_move,
+    result,
+    score,
+)
+from gym.spaces import Box, Discrete
+from pygame import Surface
+from render import render_game
+from state import State
+from torch import nn
+
+HERE = Path(__file__).parent.resolve()
+
+# Hack as this won't pip install on replit
+sys.path.append(str(HERE / "PettingZoo"))
+
+
+ALL_POSSIBLE_MOVES = np.arange(BOARD_SIZE**2 + 1)
+
+# Visuals
+SCREEN_SIZE = (500, 500)
+
+# The komi to use is much debated. 7.5 seems to
+# generalise well for different board sizes
+# lifein19x19.com/viewtopic.php?f=15&t=17750
+# 7.5 is also the komi used in alpha-go vs Lee Sedol
+# (non-integer means there are no draws)
+
+KOMI = 7.5
+
+
+def transition_function(state: State, action: int) -> State:
```

Comment:
A short docstring would be helpful here too - just reduces the amount of uncertainty & thinking required